implementation/interfacing_with_emlib.tex

```diff
@@ -164,32 +164,226 @@ \subsection{Naming Conventions}
+A list of examples that demonstrates how the bindings work is shown in \autoref{tab:emlib_examples}, these are either written from scratch, or directly ported from {\emlib}'s examples.
+
+\begin{table}[H]
+  \centering
+  \begin{tabular}{r|p{10cm}}
+    \textbf{Module} & \textbf{Purpose} \\
+    \hline
+\prog{cmu}     &
+The \gls{cmu} provides functions to manage the different clocks and oscillators on the {\gecko}.
+This module is necessary in order to configure the clocks that are required by other peripherals to function. \\
+
+\prog{dma}     &
+We wanted to use \gls{dma} for the {\prog{sensor-tracker}} application because it can be used to transfer data without intercepting the \gls{cpu}. \\
+
+\prog{ebi}     &
+The \gls{ebi} is used in the {\gecko}'s internal memory map, and simplifies the process of writing data to e.g. the LCD on the {\chip{DK}}. \\
+
+\prog{emu}     &
+The \gls{emu} module controls the different energy modes on the {\gecko}.
+We needed this functionality for the {\prog{sensor-tracker}} application. \\
+
+\prog{gpio}    &
+The \gls{gpio} was one of the first modules to be ported, it is used extensively throughout the bindings and in the applications. \\
+
+\prog{lesense} &
+The \gls{lesense} can be configured to automatically collect data from multiple sensors, we needed this for the {\prog{sensor-tracker}}. \\
+
+\prog{acmp}    &
+The \gls{acmp} was ported alongside \gls{lesense}, it is used to compare two analog signals and tell which one is greater. \\
+
+\prog{adc}     &
+The \gls{adc} was needed by the {\prog{sensor-tracker}}, it has been used to get the internal temperature of the \gls{cpu}. \\
+
+\prog{usart}   &
+Primarily we wanted the \gls{usart} for easy debugging and I/O from a computer. \\
+
+\prog{leuart}  &
+The \gls{leuart} was ported a while after the \gls{usart}.
+It has the same functionality, but it works with a lower-frequency clock and at requires less energy than the \gls{usart}. \\
+
+\prog{i2c}     &
+The \gls{i2c} has many of the same use cases as the different \gls{uart} types, but it works at lower energy levels.
+It is used in by the {\prog{sensor-tracker}}. \\
+
+\prog{rtc}     &
+The \gls{rtc} was an easy module to write bindings for, it is used in several examples for timing purposes, as well as in the {\prog{sensor-tracker}}. \\
+
+\prog{timer}   &
+The Timer was one of the first modules to be ported, it worked as a proof-of-concept for the design of the bindings, as described earlier in this section. \\
+
+    \hline
+  \end{tabular}
+
+  \caption{Peripheral bindings for {\emlib}.}
+  \label{tab:emlib_peripheral_bindings}
+\end{table}
+
+
+\begin{table}[H]
+  \centering
+  \begin{tabular}{r|p{7cm}|p{2.1cm}}
+    \textbf{Example} & \textbf{Purpose} & \textbf{Uses} \\
+    \hline
+\prog{buttons\_int}  &
+Demonstrates interrupts by lighting a led on the {\chip{STK}} when the respective button has been pressed. &
+\gls{gpio}, IRQ \\
```

Comment:
The (`Uses`) column seems to be almost references to the modules in the previous table. The should have a matching cases and their meaning should either be more explicit or described.